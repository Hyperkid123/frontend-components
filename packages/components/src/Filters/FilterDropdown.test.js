import React from 'react';
import FilterDropdown from './FilterDropdown';
import { render } from '@testing-library/react';

describe('FilterDropdown component', () => {
  const defaultProps = {
    filters: {},
    addFilter: jest.fn(),
    removeFilter: jest.fn(),
    filterCategories: [{ title: '', type: '', urlParam: '', values: [{ label: '', value: '' }] }],
  };

  it('should render', () => {
    const { container } = render(<FilterDropdown {...defaultProps} />);
    expect(container).toMatchSnapshot();
  });

  it('should render with a component as label', () => {
    const { container } = render(<FilterDropdown {...defaultProps} label={<React.Fragment />} />);
    expect(container).toMatchSnapshot();
  });
});
