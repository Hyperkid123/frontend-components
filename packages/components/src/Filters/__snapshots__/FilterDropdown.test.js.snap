// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FilterDropdown component should render 1`] = `
<Select
  className="ins-c-filter__dropdown"
  isOpen={false}
  onOpenChange={[Function]}
  ouiaId="OUIA-Generated-RHI/FilterDropdown-1"
  ouiaSafe={true}
  toggle={[Function]}
>
  <SelectGroup
    key="0"
    label=""
  >
    <SelectList>
      <SelectOption
        key="check00"
      >
        <FilterInput
          addRemoveFilters={[Function]}
          aria-label=""
          filters={{}}
          id="check00"
          label=""
          param=""
          type=""
          value=""
        />
      </SelectOption>
    </SelectList>
  </SelectGroup>
</Select>
`;

exports[`FilterDropdown component should render with a component as label 1`] = `
<Select
  className="ins-c-filter__dropdown"
  isOpen={false}
  onOpenChange={[Function]}
  ouiaId="OUIA-Generated-RHI/FilterDropdown-1"
  ouiaSafe={true}
  toggle={[Function]}
>
  <SelectGroup
    key="0"
    label=""
  >
    <SelectList>
      <SelectOption
        key="check00"
      >
        <FilterInput
          addRemoveFilters={[Function]}
          aria-label=""
          filters={{}}
          id="check00"
          label=""
          param=""
          type=""
          value=""
        />
      </SelectOption>
    </SelectList>
  </SelectGroup>
</Select>
`;
